src/Host/Client/Impl/Session/RSession.cs

```diff
@@ -454,14 +461,10 @@ internal sealed class RSession : IRSession, IRCallbacks {
         /// Graph app may call Win32 API directly rather than going via R API callbacks.
         /// </summary>
         /// <returns>Pressed button code</returns>
-        async Task<MessageButtons> IRCallbacks.ShowDialog(IReadOnlyList<IRContext> contexts, string s, bool isEvaluationAllowed, MessageButtons buttons, CancellationToken ct) {
+        async Task<MessageButtons> IRCallbacks.ShowDialog(IReadOnlyList<IRContext> contexts, string s, MessageButtons buttons, CancellationToken ct) {
```

Comment:
Absolutely. That's the whole point of the change (and the corresponding host change) - that evals can be fired at any moment, and will be serviced ASAP in an asynchronous fashion.

It's also why I want to keep `BeginEvaluationAsync` and friends - this lets you do evaluations in a controlled context, and check the context before running the eval. However, we have  evals all around that don't actually need this, and that suffer from the constraints that `BeginEvaluationAsync` model with the queue imposes - setting/removing breakpoints was one example of that, but I'm sure there are more similar cases.